consensus/core/src/network/tonic_network.rs

```diff
@@ -519,4 +737,60 @@ mod test {
             test_block_0.serialized(),
         );
     }
+
+    #[tokio::test]
+    async fn tonic_subscribe_blocks() {
+        let (context, keys) = Context::new_for_test(4);
+
+        let context_0 = Arc::new(
+            context
+                .clone()
+                .with_authority_index(context.committee.to_authority_index(0).unwrap()),
+        );
+        let mut manager_0 = TonicManager::new(context_0.clone());
+        let client_0 = <TonicManager as NetworkManager<Mutex<TestService>>>::client(&manager_0);
+        let service_0 = Arc::new(Mutex::new(TestService::new()));
+        manager_0
+            .install_service(keys[0].0.clone(), service_0.clone())
+            .await;
+
+        let context_1 = Arc::new(
+            context
+                .clone()
+                .with_authority_index(context.committee.to_authority_index(1).unwrap()),
+        );
+        let mut manager_1 = TonicManager::new(context_1.clone());
+        let client_1 = <TonicManager as NetworkManager<Mutex<TestService>>>::client(&manager_1);
+        let service_1 = Arc::new(Mutex::new(TestService::new()));
+        manager_1
+            .install_service(keys[1].0.clone(), service_1.clone())
+            .await;
+
+        let client_0_round = 50;
+        let receive_stream_0 = client_0
+            .subscribe_blocks(
+                context_0.committee.to_authority_index(1).unwrap(),
+                client_0_round,
+                Duration::from_secs(5),
+            )
+            .await
+            .unwrap();
+        receive_stream_0
+            .enumerate()
+            .for_each_concurrent(None, |(i, item)| async move {
+                assert_eq!(item, block_for_round(client_0_round + i as Round + 1));
+            })
+            .await;
```

Comment:
Maybe we want to confirm that this stream has indeed received the expected number of blocks? This stream might not return anything and the test will still pass